base/sparse/abstractsparse.jl

```diff
@@ -14,3 +16,21 @@ issparse(A::AbstractArray) = false
 issparse(S::AbstractSparseArray) = true
 
 indtype{Tv,Ti}(S::AbstractSparseArray{Tv,Ti}) = Ti
+
+function union!{Tv,Ti,N}(s::AbstractSet, x::AbstractSparseArray{Tv,Ti,N})
+    vals = nonzeros(x)
+    length(vals) == length(x) ? union!(s, vals) : union!(s, [zero(Tv); vals])
+end
+
+# assumes that AbstractSparseArray type stores nonzeros in column major order
```

Comment:
Should probably only be defining a method that's built on this assumption for concrete sparse types that are known to be column major. The only other concrete subtype in Base looks to be `CHOLMOD.Sparse`, and I'm not sure whether that is always column major. It doesn't have a `nonzeros` method so this wouldn't really work right now anyway...